fog/sql_recovery_db/src/lib.rs

```diff
@@ -590,10 +590,7 @@ impl SqlRecoveryDb {
                     let proto_ingested_block_data = ProtoIngestedBlockData {
                         e_tx_out_records: txs.to_vec(),
                     };
-                    let mut bytes =
-                        Vec::<u8>::with_capacity(proto_ingested_block_data.encoded_len());
-                    proto_ingested_block_data.encode(&mut bytes)?;
-                    bytes
+                    proto_ingested_block_data.encode_to_vec()
```

Comment:
The `encode_to_vec()` version is infallible and thus allowed for the removal of the `EncodeError`